src/WebUI/Components/DaemonSetListingComponent.tsx

```diff
@@ -0,0 +1,141 @@
+import * as Resources from "../Resources";
+import "./DaemonSetListingComponent.scss";
+import { ListComponent } from "./ListComponent";
+import { IVssComponentProperties } from "../Types";
+import { Ago } from "azure-devops-ui/Ago";
+import { Utils } from "../Utils";
+
+const setNameKey = "set-name-key";
+const imageKey = "image-key";
+const podsKey = "pods-key";
+const ageKey = "age-key";
+
+export interface IDaemonSetComponentProperties extends IVssComponentProperties {
+    daemonSetList: V1DaemonSetList;
+    onItemInvoked?: (item?: any, index?: number, ev?: Event) => void;
+}
+
+
+export class DaemonSetListingComponent extends BaseComponent<IDaemonSetComponentProperties, {}> {
+    public render(): React.ReactNode {
+        return (
+            <div>{
+                this.props.daemonSetList &&
+                this.props.daemonSetList.items &&
+                <ListComponent
+                    className={css("ds-list-view", "depth-16")}
+                    items={this.props.daemonSetList.items}
+                    columns={DaemonSetListingComponent._getColumns()}
+                    onRenderItemColumn={DaemonSetListingComponent._onRenderItemColumn}
+                />
+            }</div>
+        );
+    }
+
+    // private _openDeploymentItem = (item?: any, index?: number, ev?: Event) => {
+    //     if (this.props.onItemInvoked) {
+    //         this.props.onItemInvoked(item, index, ev);
+    //     }
+    // }
+
+    private static _getColumns(): IColumn[] {
+        let columns: IColumn[] = [];
+        const headerColumnClassName: string = "ds-secondary-text";
+
+        columns.push({
+            key: setNameKey,
+            name: Resources.DaemonSetText,
+            fieldName: setNameKey,
+            minWidth: 140,
+            maxWidth: 140,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: imageKey,
+            name: Resources.ImageText,
+            fieldName: imageKey,
+            minWidth: 160,
+            maxWidth: 160,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: podsKey,
+            name: Resources.PodsText,
+            fieldName: podsKey,
+            minWidth: 80,
+            maxWidth: 80,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: ageKey,
+            name: Resources.AgeText,
+            fieldName: ageKey,
+            minWidth: 80,
+            maxWidth: 80,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+
+        return columns;
+    }
+    private static _onRenderItemColumn(daemonSet?: V1DaemonSet, index?: number, column?: IColumn): React.ReactNode {
```

Comment:
shall we make sure ellipses are shown for columns if they are bigger than column and also the tooltip